packages/client/wallets/smart-wallet/src/blockchain/wallets/account/config.ts

```diff
@@ -1,29 +1,72 @@
-import type { Address } from "viem";
-
 import type { CrossmintWalletService } from "../../../api/CrossmintWalletService";
 import type { SmartWalletChain } from "../../../blockchain/chains";
 import { SmartWalletError } from "../../../error";
-import type { SupportedEntryPointVersion, SupportedKernelVersion } from "../../../types/internal";
+import type {
+    PreExistingWalletProperties,
+    SupportedEntryPointVersion,
+    SupportedKernelVersion,
+} from "../../../types/internal";
 import type { UserParams } from "../../../types/params";
-import type { SignerData } from "../../../types/service";
+import type { SignerData, SmartWalletConfig } from "../../../types/service";
+import { AccountConfigCache } from "./cache";
 import { EOASignerConfig, PasskeySignerConfig, type SignerConfig } from "./signer";
 
-export class AccountConfigFacade {
-    constructor(private readonly crossmintService: CrossmintWalletService) {}
+interface AccountConfig {
+    entryPointVersion: SupportedEntryPointVersion;
+    kernelVersion: SupportedKernelVersion;
+    userWithId: UserParams & { id: string };
+    existing?: PreExistingWalletProperties;
+}
+export class AccountConfigService {
+    constructor(
+        private readonly crossmintService: CrossmintWalletService,
+        private readonly configCache: AccountConfigCache
+    ) {}
 
     public async get(
         user: UserParams,
         chain: SmartWalletChain
     ): Promise<{
+        config: AccountConfig;
+        cached: boolean;
+    }> {
+        const cached = this.configCache.get(user);
+        if (cached != null) {
+            return {
+                config: this.validateAndFormat(user, cached),
+                cached: true,
+            };
+        }
+
+        const config = await this.crossmintService.getSmartWalletConfig(user, chain);
+        return { config: this.validateAndFormat(user, config), cached: false };
+    }
+
+    public cache({
+        entryPointVersion,
+        kernelVersion,
+        user,
+        existing,
+    }: {
         entryPointVersion: SupportedEntryPointVersion;
         kernelVersion: SupportedKernelVersion;
-        userId: string;
-        existingSignerConfig?: SignerConfig;
-        smartContractWalletAddress?: Address;
-    }> {
-        const { entryPointVersion, kernelVersion, signers, smartContractWalletAddress, userId } =
-            await this.crossmintService.getSmartWalletConfig(user, chain);
+        user: UserParams & { id: string };
+        existing: PreExistingWalletProperties;
+    }) {
+        this.configCache.clear();
```

Comment:
Clearing the cache here is the simplest way to keep it tidy, at the expense of not caching wallets for multiple users wallets.

Since that's not a use case we expect atm, opting for simplicity.